source/s3_copy_object.c

```diff
@@ -397,18 +398,15 @@ static int s_s3_copy_object_prepare_request(struct aws_s3_meta_request *meta_req
                 return AWS_OP_ERR;
             }
 
-            size_t part_size = (size_t)part_size_uint64;
-
-            const size_t MIN_PART_SIZE = 64L * 1024L * 1024L; /* minimum partition size */
-            if (part_size < MIN_PART_SIZE) {
-                part_size = MIN_PART_SIZE;
+            uint64_t max_part_size = GB_TO_BYTES((uint64_t)5);
+            if (max_part_size > SIZE_MAX) {
+                max_part_size = SIZE_MAX;
             }
+            uint32_t num_parts = 0;
+            size_t part_size = 0;
 
-            uint32_t num_parts = (uint32_t)(copy_object->synced_data.content_length / part_size);
-
-            if ((copy_object->synced_data.content_length % part_size) > 0) {
-                ++num_parts;
-            }
+            aws_s3_calculate_optimal_mpu_part_size_and_num_parts(
+                copy_object->synced_data.content_length, s_min_copy_part_size, max_part_size, &part_size, &num_parts);
```

Comment:
Do we ever mention in the public doc that there is a min_copy_part_size? It will override the part size set from user, right?

Reply:
we never documented it and it was internal constant ever since we introduced this class. 
for copy it needs to be quite a bit bigger than client part size and we probably want to expose separate config for it. 
added todo